pkg/kubectl/resource_printer.go

```diff
@@ -277,16 +275,35 @@ func (h *HumanReadablePrinter) printHeader(columnNames []string, w io.Writer) er
 	return nil
 }
 
-func formatEndpoints(endpoints []api.Endpoint) string {
-	if len(endpoints) == 0 {
+func formatEndpoints(endpoints *api.Endpoints) string {
+	if len(endpoints.Subsets) == 0 {
 		return "<none>"
 	}
 	list := []string{}
-	for i := range endpoints {
-		ep := &endpoints[i]
-		list = append(list, net.JoinHostPort(ep.IP, strconv.Itoa(ep.Port)))
+	max := 3
+	more := false
+Loop:
+	for i := range endpoints.Subsets {
+		ss := &endpoints.Subsets[i]
+		for i := range ss.Ports {
+			port := &ss.Ports[i]
+			if port.Name == "" { // TODO: add multi-port support.
+				for i := range ss.Addresses {
+					if len(list) == max {
+						more = true
+						break Loop
+					}
+					addr := &ss.Addresses[i]
+					list = append(list, fmt.Sprintf("%s:%d", addr.IP, port.Port))
+				}
+			}
+		}
+	}
+	ret := strings.Join(list, ",")
+	if more {
+		ret += "..."
```

Comment:
Correct.  I consulted with Sam and he thought that was a good compromise, rather than printing an arbitrary list.  I could change 3 to 5 if people care.